deepchecks/base/dataset.py

```diff
@@ -781,109 +783,213 @@ def select(
         else:
             return self.copy(new_data)
 
-    def validate_shared_features(self, other) -> t.List[Hashable]:
+    @classmethod
+    def ensure_not_empty_dataset(cls, obj: t.Any, *, cast: bool = False) -> 'Dataset':
+        """TODO: add comments"""
+        if isinstance(obj, Dataset):
+            if len(obj.data) == 0:
+                raise DeepchecksValueError('dataset cannot be empty')
+            return obj
+        
+        elif cast and isinstance(obj, pd.DataFrame):
+            if len(obj) == 0:
+                raise DeepchecksValueError('dataframe cannot be empty')
+            return Dataset(obj)
+        
+        # TODO: should we add case for the numpy arrays?
+        
+        else:
+            value_type = type(obj).__name__
+            if cast:
+                raise DeepchecksValueError(f'cannot cast instance of {value_type} into Dataset object')
+            else:
+                raise DeepchecksValueError(f'non-empty Dataset instance was expected, instead got {value_type}')
+    
+    @classmethod
+    def datasets_share_features(cls, datasets: t.List['Dataset']) -> bool:
         """
-        Return the list of shared features if both datasets have the same feature column names. Else, raise error.
+        Verify that all provided datasets share same features.
 
         Args:
-            other: Expected to be Dataset type. dataset to compare features list
+            datasets (List[Dataset]): list of datasets to validate
 
         Returns:
-            List[Hashable] - list of shared features names
+            bool: True if all datasets share same features, otherwise False
 
         Raises:
-            DeepchecksValueError if datasets don't have the same features
-
+            AssertionError:
+                'datasets' parameter is not instance of a list;
+                'datasets' is empty or contains only one dataset;
         """
-        Dataset.validate_dataset(other)
-        if sorted(self.features) == sorted(other.features):
-            return list(self.features)
-        else:
-            raise DeepchecksValueError('Check requires datasets to share the same features')
+        assert isinstance(datasets, list)
```

Comment:
not sure I like the use of assert, what happens if it fails?

Reply:
it is common in the python world to use assert to verify that input parameters of the function satisfy its contract (signature). Ff `datasets` is not a list then `AssertionError` will be raised which means that function is used improperly 

I will add a message to that assert to make it cleaner what happened

Reply:
oh, also regarding this

> ..not sure about the classmethods ..

they do not depend upon instance variables (state), so I decided to declare thems as classmethods